File: python_roguelike/data/relic_data.py
from dataclasses import dataclass, field
from typing import List
from .effect_data import EffectData


@dataclass
class RelicData:
    id: str = ""
    name: str = ""
    description: str = ""
    star_rating: int = 1
    is_boss_relic: bool = False
    effects: List[EffectData] = field(default_factory=list)

File: src/Roguelike.Core/Core/Data/RelicData.cs
using System.Collections.Generic;

namespace Roguelike.Data
{


    public class RelicData
    {


        public string Id { get; set; }


        public string Name { get; set; }


        public string Description { get; set; }


        public int StarRating { get; set; }


        public bool IsBossRelic { get; set; } = false;


        public List<EffectData> Effects { get; set; } = new List<EffectData>();


        public RelicData() { }
    }
}
